src/init.cpp

```diff
@@ -1487,6 +1492,8 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
 
                 bool is_coinsview_empty = fReset || fReindexChainState || pcoinsTip->GetBestBlock().IsNull();
                 if (!is_coinsview_empty) {
+                    LOCK(cs_main);
```

Comment:
This seems unrelated. Maybe leave it for #11596 instead? Or just do as I suggested at https://github.com/bitcoin/bitcoin/pull/11596#discussion_r155811677 and put a cs_main at the top of the load-chain-data block here.

Reply:
It's needed for the assert below (L1502).

Reply:
Looks like you can remove this line now?